### ISSUE TITLE
Official mtproto proxy specification

### ISSUE THREAD

**Issue by p1ratrulezzz** [TelegramMessenger/MTProxy#21]
Does it exist? There are many proxy implementations already but they all build just from experiments and some of them doesn't work on android clients at all. Is there any official document that describes how to implement your own mtproto proxy? The packet flow actually is the same as for mtproto, but there are some differences and also it is not clear what to do with proxy tag for your custom implementation

**Comment by makkarpov** [comment]
And provided that C code is fucking unreadable, protocol specs worth a lot.

**Comment by euphoria360** [comment]
Agreed. It would be great to see protocol specs and the way it works!
How exactly are we getting connected to telegram servers via this proxy?
What are those servers written in proxy-multi.conf?

**Comment by VictoriaRaymond** [comment]
As the owner of [V2Ray](https://github.com/v2ray/v2ray-core) (a widely used proxy software), I appreciate if Telegram publishes the spec of MTProxy. V2Ray is planning to support MTProxy, in aid of other existing proxies, such as socks5 and HTTP.

**Comment by yulewang** [comment]
@VictoriaRaymond I am very excited about V2Ray supported MTProxy.
But it can't seem to pass the "tag" for the promotion channel
